jib-plugins-common/src/test/java/com/google/cloud/tools/jib/plugins/common/UpdateCheckerTest.java

```diff
@@ -0,0 +1,201 @@
+    FileTime modifiedTime = FileTime.from(Instant.now().minusSeconds(12));
+    setupConfigAndLastUpdateCheck();
+    Files.write(
+        configDir.resolve("lastUpdateCheck"),
+        modifiedTime.toString().getBytes(StandardCharsets.UTF_8));
+    Files.setLastModifiedTime(configDir.resolve("lastUpdateCheck"), modifiedTime);
+    Optional<String> message =
+        UpdateChecker.performUpdateCheck(false, "1.0.2", testWebServer.getEndpoint(), configDir);
+    Assert.assertFalse(message.isPresent());
+
+    // lastUpdateCheck should not have changed
+    String lastUpdateTime =
+        new String(
+            Files.readAllBytes(configDir.resolve("lastUpdateCheck")), StandardCharsets.UTF_8);
+    Assert.assertEquals(Instant.parse(lastUpdateTime), modifiedTime.toInstant());
+  }
+
+  @Test
+  public void testPerformUpdateCheck_skip() {
+    Optional<String> message =
+        UpdateChecker.performUpdateCheck(true, "1.0.2", testWebServer.getEndpoint(), configDir);
+    Assert.assertFalse(message.isPresent());
+  }
+
+  @Test
+  public void testPerformUpdateCheck_systemProperty() {
+    System.setProperty(PropertyNames.DISABLE_UPDATE_CHECKS, "true");
+    Optional<String> message =
+        UpdateChecker.performUpdateCheck(true, "1.0.2", testWebServer.getEndpoint(), configDir);
+    Assert.assertFalse(message.isPresent());
+  }
+
+  @Test
+  public void testPerformUpdateCheck_configDisabled() throws IOException {
+    UpdateChecker.ConfigJsonTemplate config = new UpdateChecker.ConfigJsonTemplate();
+    config.setDisableUpdateCheck(true);
+    JsonTemplateMapper.writeTo(config, Files.newOutputStream(configDir.resolve("config.json")));
+
+    Optional<String> message =
+        UpdateChecker.performUpdateCheck(true, "1.0.2", testWebServer.getEndpoint(), configDir);
+    Assert.assertFalse(message.isPresent());
+  }
+
+  @Test
+  public void testPerformUpdateCheck_failSilently()
+      throws InterruptedException, GeneralSecurityException, URISyntaxException, IOException {
+    try (TestWebServer badServer = new TestWebServer(false, Collections.singletonList("bad"), 1)) {
+      Optional<String> message =
+          UpdateChecker.performUpdateCheck(true, "1.0.2", badServer.getEndpoint(), configDir);
+      Assert.assertFalse(message.isPresent());
+    }
+  }
+
+  @Test
+  public void testFinishUpdateCheck_success() throws InterruptedException {
+    ExecutorService executorService = Executors.newSingleThreadExecutor();
+    UpdateChecker updateChecker =
+        new UpdateChecker(executorService.submit(() -> Optional.of("Hello")));
+    Thread.sleep(100);
+    Optional<String> result = updateChecker.finishUpdateCheck();
+    Assert.assertTrue(result.isPresent());
+    Assert.assertEquals("Hello", result.get());
+  }
+
+  @Test
+  public void testFinishUpdateCheck_notDone() {
+    ExecutorService executorService = Executors.newSingleThreadExecutor();
+    UpdateChecker updateChecker =
```

Comment:
`Future` is a simple interface, so I think it's better to implement one whose `isDone()` always returns false.